qemu/tests/chardev_tls_encryption.py

```diff
@@ -0,0 +1,94 @@
+        hostname = process.run('hostname', ignore_status=False, shell=True,
+                               verbose=True).stdout_text.strip()
+        port = str(utils_misc.find_free_ports(5000, 9999, 1, hostname)[0])
+
+        # Scenario 1
+        gnutls_cmd_server = params.get("gnutls_cmd_server")
+        if gnutls_cmd_server:
+            gnutls_cmd_server = gnutls_cmd_server % port
+            params["extra_params"] = params["extra_params"] % (hostname, port)
+            tls_server = aexpect.run_bg(gnutls_cmd_server)
+            params['start_vm'] = 'yes'
+            vm_name = params['main_vm']
+            env_process.preprocess_vm(test, params, env, vm_name)
+            if not utils_misc.wait_for(
+                    lambda: "Channel binding 'tls-unique'" in
+                            tls_server.get_output(), first=5, timeout=15):
+                test.fail("tls server can't connect client succssfully.")
+
+        # Scenario 2
+        gnutls_cmd_client = params.get("gnutls_cmd_client")
+        if gnutls_cmd_client:
+            gnutls_cmd_client = gnutls_cmd_client % (port, hostname)
+            params["extra_params"] = params["extra_params"] % (hostname, port)
+            params['start_vm'] = 'yes'
+            vm_name = params['main_vm']
+            env_process.preprocess_vm(test, params, env, vm_name)
+            tls_client = aexpect.run_bg(gnutls_cmd_client)
+            if not utils_misc.wait_for(lambda: "Simple Client Mode:" in
+                    tls_client.get_output(), first=5, timeout=15):
+                test.fail("tls client can't connect server succssfully.")
+
+        # Scenario 3:
+        guest_cmd = params.get("guest_cmd")
+        if guest_cmd:
+            params["start_vm"] = "yes"
+            vms = params.get("vms").split()
+            params["extra_params"] = params["extra_params_vm1"] % (hostname,
+                                                                   port)
+            env_process.preprocess_vm(test, params, env, vms[0])
+            vm1 = env.get_vm(vms[0])
+            session_vm1 = vm1.wait_for_login()
+            session_vm1.cmd(guest_cmd)
+            params["extra_params"] = params["extra_params_vm2"] % (hostname,
+                                                                   port)
+            env_process.preprocess_vm(test, params, env, vms[1])
+            vm2 = env.get_vm(vms[1])
+            if not utils_misc.wait_for(lambda: "BOOT_IMAGE" in
+                    session_vm1.cmd_output("cat /root/client-log"),
+                                       first=5, timeout=15):
+                test.fail("Can't connect the tls client inside tls server guest.")
+            session_vm1.cmd("rm /root/client-log -rf")
+            session_vm1.close
+            vm1.destroy()
+            vm2.destroy()
+    finally:
+        gnutls_pid = process.getoutput("pgrep -f gnutls", shell=True)
+        if gnutls_pid:
+            process.run("pkill gnutls")
```

Comment:
Use `-9` ?

Reply:
> Use `-9` ?

using '-9' will raise some error and seems 'pkill' is more flexible than ‘kill’, this is ok.

Reply:
> > Use `-9` ?
> 
> using '-9' will raise some error and seems 'pkill' is more flexible than ‘kill’, this is ok.

I mean `pkill -9`,  not `kill -9`.   What's the error when use pkill ?

Reply:
it works well, thanks and update